### DISCUSSION TITLE
[Hiring] Is it possible to hire someone via here? (for a Model Caster with Spatie DTO)

### DISCUSSION THREAD

**Discussion by basepack** [livewire/livewire#2332] [1. Help]
Hi,

I am looking for an urgent solution for something pretty unusual, but when we have the solution created I will post it here too so the community can benefit from this.

Essential I want to create the following:
- Using a Spatie DTO **Item** with JSON casting from https://github.com/jessarcher/laravel-castable-data-transfer-object
```
class DayData extends DataTransferObject 
{
    public string $day;
    public bool $active;
}

```
- That Item within a Spatie DTO **Collection**
```
class DayDataCollection extends DataTransferCollection
{
    public function current(): DayData
    {
        return parent::current();
    }
}
```

On a model somewhere:
```
class SomeModel extends Model
{
    protected $casts = [
        'days'                => DayDataCollection::class, 
    ];
}
```

And then the culprit where I am stuck, a Livewire component:
```
class SomeComponent extends Component
{
    public $rules = [
        'someModel.days.*.day'                    => 'string',
        'someModel.days.*.active'                 => 'boolean',
   ];
}
```

The desired outcome would be that we can PR this to jessarchers repo as I can imagine this is quite handy for other folks too. 

I have no idea how to proceed from now on for hiring part, but everyone with a working solution (you show it to me) can get paid and then you send me the code. You then should PR this after the payment to the @jessarcher's repo too as there is already a issue about it too: https://github.com/jessarcher/laravel-castable-data-transfer-object/issues/4

Edit: I should mention that I have it working until the Livewire part. So essential the DTO's are as a Collection Castable to Json back and forth. However the last part with the Livewire Rules actually won't work. Currently I have fixed it for now like this:
```
class SomeComponent extends Component
{
    public $days;

    public function mount()
    {
         $this->days  = $this->someModel->days->toArray();
    }

    public function save()
    {
         $this->someModel->days  = DayDataCollection::create($this->days);
    }
}
```

**Comment by IP-Developer** [comment]
What is the content of `$this->days` when calling `save()` ?

**Reply by basepack** [reply]
Just the array: [{"day": "monday", "active": true}, {"day": "tuesday", "active": false}, etc... }]

**Reply by IP-Developer** [reply]
It's not very clear what exactly you are trying to achieve. If you want to always have a collection of DTOs in your Livewire component, you could use hydrated/dehydrated hooks. Basically something like this: 
```php
hydrateDays($array) 
{
	$this->days = DayDataCollection::create($this->days);
	
	//iterate through the array and create DayData instance for each value, if needed
}

dehydrateDays() 
{
	return $this->days->toArray();
	
	//iterate through the array and cast each DayData instance to array, if needed
}
```

P.S. A Laravel Playground with an example of your current solution would help speed this up.

**Reply by basepack** [reply]
@IP-Developer thanks for your help! I have in my initial post first said what I want to archive. After that, you see and `Edit:`, and how I have currently sort of have my app working. 

So my first intention was to have it working like this:
```
// Model
protected $casts = [
  'days'  => DayDataCollection::class,  
];

// Livewire component
public $rules = [
    'someModel.days.*.day'                    => 'string',
    'someModel.days.*.active'                 => 'boolean',
];
```

And this bit is not working as it throws diverse errors whatever I try.

**Reply by basepack** [reply]
If you have a solution we can make an agreement for that as I proposed. However for now I removed the custom JSON casters in `$casts` to be just `'days' => 'array'` because I really do not like the additional overhead in my app by being forced to write around a half-working solution constantly.

I do still use `DayDataCollection::create($this->days);` here and there in my code for sake of easy use.
